agent/agent.go

```diff
@@ -1,18 +1,20 @@
 package agent
 
 import (
-	"errors"
+	"fmt"
 	"time"
 
 	"github.com/pathfinder-cm/pathfinder-agent/daemon"
+	"github.com/pathfinder-cm/pathfinder-agent/metrics"
 	"github.com/pathfinder-cm/pathfinder-agent/util"
 	"github.com/pathfinder-cm/pathfinder-go-client/pfclient"
 	"github.com/pathfinder-cm/pathfinder-go-client/pfmodel"
 	log "github.com/sirupsen/logrus"
 )
 
 type Agent interface {
-	Run()
+	Run(agentType string)
+	RunMetrics() error
```

Comment:
Instead of doing this, you should make `agent.go` containing only the interface with method `Run()` then you can create implementation `provision_agent.go` and `metrics_agent.go`.